lib/cartopy/util.py

```diff
@@ -99,3 +101,298 @@ def add_cyclic_point(data, coord=None, axis=-1):
     else:
         return_value = new_data, new_coord
     return return_value
+
+
+def _add_cyclic_data(data, axis=-1):
+    """
+    Add a cyclic point to a data array.
+
+    Parameters
+    ----------
+    data : ndarray
+        An n-dimensional array of data to add a cyclic point to.
+    axis: int, optional
+        Specifies the axis of the data array to add the cyclic point to.
+        Defaults to the right-most axis.
+
+    Returns
+    -------
+    The data array with a cyclic point added.
+    """
+    slicer = [slice(None)] * data.ndim
+    try:
+        slicer[axis] = slice(0, 1)
+    except IndexError:
+        estr = 'The specified axis does not correspond to an array dimension.'
+        raise ValueError(estr)
+    npc = np.ma if np.ma.is_masked(data) else np
+    return npc.concatenate((data, data[tuple(slicer)]), axis=axis)
+
+
+def _add_cyclic_lon(lon, axis=-1, cyclic=360):
+    """
+    Add a cyclic point to a longitude array.
+
+    Parameters
+    ----------
+    lon: ndarray, optional
+        An array which specifies the coordinate values for
+        the dimension the cyclic point is to be added to.
+    axis: int, optional
+        Specifies the axis of the longitude array to add the cyclic point to.
+        Defaults to the right-most axis.
+    cyclic: float, optional
+        Width of periodic domain (default: 360)
+
+    Returns
+    -------
+    The coordinate `lon` with a cyclic point added.
+    """
+    npc = np.ma if np.ma.is_masked(lon) else np
+    # get cyclic longitudes
+    # clon is the code from basemap (addcyclic)
+    # https://github.com/matplotlib/basemap/blob/master/lib/mpl_toolkits/basemap/__init__.py
+    clon = (np.take(lon, [0], axis=axis) +
+            cyclic * np.sign(np.diff(np.take(lon, [0, -1], axis=axis),
+                                     axis=axis)))
+    # basemap ensures that the values do not exceed cyclic
+    # (next code line). We do not do this to deal with rotated grids that
+    # might have values not exactly 0.
+    #     clon = npc.where(clon <= cyclic, clon, np.mod(clon, cyclic))
+    return npc.concatenate((lon, clon), axis=axis)
+
+
+def _has_cyclic(lon, axis=-1, cyclic=360, prec=1e-4):
+    """
+    Check if longitudes already have a cyclic point.
+
+    Checks all differences between the first and last
+    longitudes along `axis` to be less than `prec`.
+
+    Parameters
+    ----------
+    lon: ndarray, optional
```

Comment:
```suggestion
    lon : ndarray
```

numpydoc should have a space on each side of the colon. This applies to all of the docstrings here.